server/src/main/java/org/opensearch/indices/replication/checkpoint/PublishCheckpointAction.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.replication.checkpoint;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.replication.ReplicationResponse;
+import org.opensearch.action.support.replication.ReplicationTask;
+import org.opensearch.action.support.replication.TransportReplicationAction;
+import org.opensearch.cluster.action.shard.ShardStateAction;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.index.shard.IndexShardClosedException;
+import org.opensearch.indices.IndicesService;
+import org.opensearch.node.NodeClosedException;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class PublishCheckpointAction extends TransportReplicationAction<
+    PublishCheckpointRequest,
+    PublishCheckpointRequest,
+    ReplicationResponse> {
+
+    public static final String ACTION_NAME = "indices:admin/publishCheckpoint";
+    protected static Logger logger = LogManager.getLogger(PublishCheckpointAction.class);
+
+    @Inject
```

Comment:
I'm wondering if we should also conditionally bind this component as a singleton inside of ActionModule.  With @Inject here this component will automatically be created even with the segrep feature flag off.  It will then also be registered as an available transport endpoint inside of the constructor of TransportReplicationAction.